parser/scanner.go

```diff
@@ -10,6 +10,14 @@ type Scanner struct {
 	src    string
 	slice  string
 	offset int
+
+	filename string
```

Comment:
A separate source context would be better. Something like:
```go
type Source struct {
	filename string
	src      string
}

type Scanner struct {
	src    *Source
	slice  string
	offset int
}
```
This should be a fairly harmless change, since `src` is barely used, if at all.